### PR TITLE
Move ROI to X1Y0, remove synthetic IO

### PR DESCRIPTION
**Pull request by HackerFoo**
Checking to see if I'm going in the right direction.

### CODE REVIEW

File: xc7/techmap/cells_map.v [hunk 1]
```diff
@@ -1380,37 +1380,31 @@ module RAMB36E1 (
...
+        && (x) != 72)
+
+    if(`INVALID_WIDTH_WIDE(READ_WIDTH_A))
         $error("Invalid READ_WIDTH_A: ", READ_WIDTH_A);
-    if(READ_WIDTH_B != 0
-        && READ_WIDTH_B != 1
-        && READ_WIDTH_B != 4
-        && READ_WIDTH_B != 9
-        && READ_WIDTH_B != 18
-        && READ_WIDTH_B != 36)
```
Comment (litghost) [inline xc7/techmap/cells_map.v]: This is an unrelated change?

File: xc7/archs/artix7/tiles/iopad/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+project_xray_tile(
+  PART artix7
```
Comment (litghost) [inline xc7/archs/artix7/tiles/iopad/CMakeLists.txt]: I don't imagine this actually works?

Comment (HackerFoo) [reply xc7/archs/artix7/tiles/iopad/CMakeLists.txt]: No, not yet. Do I need to define pb_type/model.xml for all SITE_TYPES? It seems that way.

File: xc7/techmap/cells_map.v (hunk 1 above)
Comment (HackerFoo) [reply xc7/techmap/cells_map.v]: Yeah, already merged.

**Comment by HackerFoo** [comment]
This might be useful for anyone looking at this PR:

### What I've changed in the prjxray scripts so far:
* form_channels
  - create synthetic IOPAD tile type
  - replace RIOI3* with IOPAD
* create_synth_tiles
  - add a map for real pads
  - manually written JSON input for now
* synth_tiles_to_pinmap_csv
  - remove single pin and no duplicate pad assertions
* assign_tile_pin_direction
  - nothing yet
* tile_import
  - add import_tiles argument to merge tiles
  - add select_y argument to select a row in the tiles
  - add generate_missing_pins argument
* physical_tile_import
  - add import_tiles argument
* arch_import
  - added tile_map support (RIOI3* --> IOPAD)
    - hardcoded for now
  - get_tiles: ignore multi-pin synth_tiles, tile_map
  - add_synthetic_tiles: split into io/constant functions
* create_edges
  - find_wire_node: return None on failure
  - connect_at: remove pin & connector loc assertions, ignore failures
  - ignore multi-pin synth tiles
* routing_import
  * import_graph_nodes
    - map IOPAD wires to RIOI3*/RIOB33 wires
    - shift gridloc for RIOB33
    - ignore when the side graph node is missing
  * add_synthetic_edges
    - ignore multi-pin synth_tiles
